### PR TITLE
extra states cobrand feature

### PR DESCRIPTION
**Pull request by struan**
This allows display strings for a state/external status combination to be configured as a cobrand feature.

For mysociety/fixmystreet-commercial#1952

[skip changelog]

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/FixMyStreet.pm [hunk 1]
```diff
@@ -310,6 +310,19 @@ sub updates_disallowed {
...
 
+sub problem_state_processed {
+    my ($self, $comment) = @_;
+
+    my $state = $comment->problem_state || '';
+    my $code = $comment->get_extra_metadata('external_status_code') || '';
+
+    my ($cfg) = $self->per_body_config('extra_state_mapping', $comment->problem);
+
+    $state = $cfg->{$state}->{$code} || $state;
```
Comment (davea) [inline perllib/FixMyStreet/Cobrand/FixMyStreet.pm:321]: This was crashing with `Modification of a read-only value attempted` if `$cfg->{$state}` was undef, so something like this is needed:

```suggestion
    $state = ( $cfg->{$state}->{$code} || $state ) if $cfg->{$state};
```

File: perllib/FixMyStreet/Cobrand/UKCouncils.pm [hunk 2]
```diff
@@ -270,6 +270,19 @@ sub relative_url_for_report {
...
 
+sub problem_state_processed {
+    my ($self, $comment) = @_;
+
+    my $state = $comment->problem_state || '';
+    my $code = $comment->get_extra_metadata('external_status_code') || '';
+
+    my $cfg = $self->feature('extra_state_mapping');
+
+    $state = $cfg->{$state}->{$code} || $state;
```
Comment (davea) [inline perllib/FixMyStreet/Cobrand/UKCouncils.pm:281]: Same as above in `FixMyStreet.pm`.

**Review by davea — CHANGES_REQUESTED** [review]
Sneaky little crash in there, otherwise looks good.

**Comment by struan** [comment]
merged in https://github.com/mysociety/fixmystreet/commit/342e235b916696da4ef7982885cefe0f8b069adf

**Pull request closed without merging**